pkg/apis/core/v1beta1/helper/condition_builder.go

```diff
@@ -90,8 +95,17 @@ func (b *defaultConditionBuilder) WithCodes(codes ...gardencorev1beta1.ErrorCode
 	return b
 }
 
+// WithClock sets a `clock.Clock` which is used for getting the current time
+func (b *defaultConditionBuilder) WithClock(clock clock.Clock) ConditionBuilder {
+	b.clock = clock
+	b.nowFunc = nil
+
+	return b
+}
+
 // WithNowFunc sets the function used for getting the current time.
-// Should only be used for tests.
+// Deprecated: Use WithClock(...) instead.
+// TODO(oliver-goetz): Remove in a future release.
 func (b *defaultConditionBuilder) WithNowFunc(now func() metav1.Time) ConditionBuilder {
```

Comment:
What about the `nowFunc` property altogether and setting the clock accordingly with the value of `now`?

Reply:
I removed `WithNowFunc` entirely now.